variable/variable_instantiate_linalg.cpp

```diff
@@ -4,31 +4,41 @@
 /// @author Simon Heybrock
 #include "scipp/core/eigen.h"
 #include "scipp/variable/structure_array_variable.tcc"
+#include "scipp/variable/structures.h"
 #include "scipp/variable/variable.h"
 
 namespace scipp::variable {
 
+// TODO For now we use hard-coded field names and offsets. The intention is to
+// generalize StructureArrayModel to support more general structures. Field
+// names and sizes/offsets would then be stored as part of the model, and would
+// be initialized dynamically at runtime.
 template <>
-constexpr auto structure_element_offset<Eigen::Vector3d> = [](scipp::index i) {
-  if (i < 0 || i > 2)
-    throw std::out_of_range("out of range index (" + std::to_string(i) +
-                            ") for element access of vector of length 3.");
-  return i;
-};
+constexpr auto structure_element_offset<Eigen::Vector3d> =
+    [](const std::string &key) {
+      static std::map<std::string, scipp::index> offsets{
+          {"x", 0}, {"y", 1}, {"z", 2}};
+      return offsets.at(key);
+    };
```

Comment:
Why did you turn this and `structure_element_offset<Eigen::Matrix3d>` into functions?
Also, why lambdas instead of regular functions?

Reply:
This was not changed here? Can't remember exaclty, I think the reason was to provide a meaningful error message, via this mechanism: https://github.com/scipp/scipp/blob/afe29e132e4e38fe59810c70d1233cca606382b7/variable/include/scipp/variable/structure_array_variable.tcc#L10-L12.